service/spi/src/main/java/org/kaazing/gateway/service/collections/MemoryTopic.java

```diff
@@ -64,14 +69,31 @@ protected Boolean initialValue() {
     private static final Logger LOGGER = LoggerFactory.getLogger(MemoryTopic.class);
 
     public MemoryTopic(String name) {
-        super(name);
+        this.name = name;
+        ;
         this.localTopicStats = new LocalTopicStatsImpl();
         if (LOGGER.isTraceEnabled()) {
             LOGGER.trace("Created topic: " + this.getName());
         }
     }
 
     @Override
+    public String getPartitionKey() {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @Override
+    public String getServiceName() {
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
     public void publish(Object o) {
```

Comment:
This should take the parameterized type: 
```
   public void publish(E o) {
```
and you should use a parameterized message:
```
Message<E> m = new Message<>(this.getName(), o, System.nanoTime(), null);
```
That may improve the handling of the wrong type case, but at least it will get rid of a compilation warning  and is more correct